### PR TITLE
  🏗️ maintenance: upgrades docker and docker-compose tooling (⚠️ devops)

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Weekly upgrade maintenance main target concerns mainly constraints on docker tooling :

 - REMOVED constraint: idna constraint since coveralls was updated to 3.0.1
 - UPGRADES constraint to **docker-compose 1.29.1** (as #2264, #2278)
 - UPGRADES all **docker-compose file configurations to [version='3.8'](https://docs.docker.com/compose/compose-file/compose-file-v3/)**
     - Passing ``--env-file`` to ``docker-compose config`` to get variable substitution right
 - UPGRADES to **docker python client 5.0.0**
 - added tests to validate docker-compose config file  and installed versions

⚠️ devops: might require installing docker-compose in nodes?
 **NOTE** You can upgrade your system docker-compose by using ``ci/github/helpers/setup_docker_compose.bash``

## How to test

```cmd
$ cd osparc-simcore
$ make info
...
 docker        : Docker version 20.10.6, build 370c289
 docker-compose: docker-compose version 1.29.1, build unknown
```

```cmd
cd tests/environment-setup
make install-dev
make tests
```

PLEASE test in your computer the following
```cmd
make devenv
source .venv/bin/activate
cd services/catalog
make install-dev
make test-dev
```
... my computer is hanging in ``services/catalog/tests/unit/with_dbs/test_entrypoint_services.py::test_director_mockup`` ... but the CI does not! ??? the error is that it cannot connect to the db and times out. ``docker logs`` on the pg reveils ... 
```cmd
LOG:  connection authorized: user=test database=test
2021-04-21 05:43:41.189 UTC [230] test@test/[unknown] FATAL:  role "test" does not exist
```
but the role DOES exist. I

### CODE REVIEW

**Comment by pcrespov** [comment]
@sanderegg , some questions regarding the ``services/docker-compose*`` configs:
- do we still need ``postgres_network`` w/o the pgbouncer?
- docker swarm still [does not support ``shm_size``](https://github.com/moby/moby/issues/26714) option. Should i drop it?

**Review by mguidon — APPROVED** [review]

**Comment by sanderegg** [comment]
> @sanderegg , some questions regarding the `services/docker-compose*` configs:
> 
> * do we still need `postgres_network` w/o the pgbouncer?
> * docker swarm still [does not support `shm_size`](https://github.com/moby/moby/issues/26714) option. Should i drop it?

no the postgres_network is not necessary anymore.

concerning shm_size, I think there is a workaround for docker swarm, I have not yet investigated it exactly:
see [this](https://stackoverflow.com/questions/55416904/increase-dev-shm-in-docker-container-in-swarm-environment-docker-stack-deploy)

I guess we can drop it in the docker-compose as it is useless in its current form.

File: services/sidecar/requirements/_test.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ aiohttp==3.7.3
     #   -c requirements/_base.txt
     #   -c requirements/_packages.txt
     #   pytest-aiohttp
-aiopg[sa]==1.2.1
+aiopg==1.2.1
```
Comment (sanderegg) [inline services/sidecar/requirements/_test.txt:12]: sa goes away? is that correct?

**Review by sanderegg — APPROVED** [review]
just if you can check the aiopg dependency in the sidecar...

File: services/sidecar/requirements/_test.txt (hunk 1 above)
Comment (pcrespov) [reply services/sidecar/requirements/_test.txt:12]: yes

**Pull request merged**